### ISSUE TITLE
Compilation warnings with GCC 8.0

### ISSUE THREAD

**Issue by frederic-mahe** [torognes/vsearch#304]
```
gcc (GCC) 8.0.1 20180128 (experimental)
Copyright (C) 2018 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.

align.cc: In function 'void nw_align(char*, char*, char*, char*, int64_t*, int64_t, int64_t, int64_t, int64_t, int64_t, int64_t, int64_t, int64_t, int64_t, int64_t, int64_t, int64_t, int64_t*, int64_t*, int64_t*, int64_t*, int64_t*, char**, int64_t, int64_t, nwinfo_s*)':
align.cc:88:14: warning: 'char* strncpy(char*, const char*, size_t)' output may be truncated copying between 1 and 10 bytes from a string of length 24 [-Wstringop-truncation]
       strncpy(*cigarendp, buf, (size_t)len);
       ~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
align.cc:88:14: warning: 'char* strncpy(char*, const char*, size_t)' output may be truncated copying between 1 and 10 bytes from a string of length 24 [-Wstringop-truncation]
       strncpy(*cigarendp, buf, (size_t)len);
       ~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
align.cc:88:14: warning: 'char* strncpy(char*, const char*, size_t)' output may be truncated copying between 1 and 10 bytes from a string of length 24 [-Wstringop-truncation]
       strncpy(*cigarendp, buf, (size_t)len);
       ~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
align.cc:88:14: warning: 'char* strncpy(char*, const char*, size_t)' output may be truncated copying between 1 and 10 bytes from a string of length 24 [-Wstringop-truncation]
       strncpy(*cigarendp, buf, (size_t)len);
       ~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
align.cc:88:14: warning: 'char* strncpy(char*, const char*, size_t)' output may be truncated copying between 1 and 10 bytes from a string of length 24 [-Wstringop-truncation]
       strncpy(*cigarendp, buf, (size_t)len);
       ~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
align.cc:105:14: warning: 'char* strncpy(char*, const char*, size_t)' output may be truncated copying between 1 and 10 bytes from a string of length 24 [-Wstringop-truncation]
       strncpy(*cigarendp, buf, (size_t)len);
       ~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```
I am going to test the new [GCC 8.1](https://gcc.gnu.org/gcc-8/changes.html) (by the way, the official webpage has specific advice for the `strncpy` warning, see also [porting to GCC 8](https://gcc.gnu.org/gcc-8/porting_to.html)).

**Comment by torognes** [comment]
Fixed in commit 5d6be05a1403eb4725fd9e020fe0aa6ffd29a282.

**Comment by frederic-mahe** [comment]
No more warnings with GCC 8.1. Awesome.

**Comment by torognes** [comment]
Fixed in version 2.8.1 just released.
